pkg/execution/planning.go

```diff
@@ -72,23 +77,43 @@ type dataSourcePlannerRef struct {
 
 func (p *planningVisitor) EnterDocument(operation, definition *ast.Document) {
 	p.operation, p.definition, p.base.Definition = operation, definition, definition
+	p.foundOperation = false
+	p.isSingleOperation = len(operation.OperationDefinitions) == 1
```

Comment:
This could be wrong in some cases.
You should instead count the number of operation.RootNodes which are of kind ast.OperationDefinition